src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    private static boolean checkOuterNotEqual(DetailAST ast) {
+        final DetailAST outerNotEqualNode = getExprChildNode(ast, TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (isNullTest(outerNotEqualNode)) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST instanceOfNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.LITERAL_INSTANCEOF);
+                if (instanceOfNode != null) {
+                    error = isTextEqual(instanceOfNode, outerNotEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    private static boolean checkOuterInstanceOf(DetailAST ast) {
+        final DetailAST outerInstanceOfNode =
+                getExprChildNode(ast, TokenTypes.LITERAL_INSTANCEOF);
+        boolean error = false;
+        if (outerInstanceOfNode != null) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST notEqualNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.NOT_EQUAL);
+                if (isNullTest(notEqualNode)) {
+                    error = isTextEqual(outerInstanceOfNode, notEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    /**
+     * To get specific Node that is child of EXPR Node.
+     *
+     * @param ast DetailAST
+     * @param token int
+     * @return DetailAST
+     */
+    private static DetailAST getExprChildNode(DetailAST ast, int token) {
+        return ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(token);
+    }
+
+    /**
+     * Judge whether this is a null test.
+     *
+     * @param ast DetailAST
+     * @return boolean
+     */
+    private static boolean isNullTest(DetailAST ast) {
+        return ast != null
+                && ast.findFirstToken(TokenTypes.LITERAL_NULL) != null;
+    }
+
+    /**
+     * To get LITERAL_IF Node that is nested.
+     *
+     * @param ast DetailAST
+     * @return nestedIfNode DetailAST
+     */
+    private static DetailAST getNestedIfNode(DetailAST ast) {
+        final DetailAST slistNode = ast.findFirstToken(TokenTypes.SLIST);
+        final DetailAST nestedIfNode;
+        if (slistNode == null) {
+            nestedIfNode = ast.findFirstToken(TokenTypes.LITERAL_IF);
+        }
+        else {
+            nestedIfNode = slistNode.findFirstToken(TokenTypes.LITERAL_IF);
+        }
+        return nestedIfNode;
+    }
+
+    /**
+     * Judge whether the text of the two nodes is the same.
+     *
+     * @param instanceOfNode DetailAST
+     * @param notEqualNode DetailAST
+     * @return error boolean
+     */
+    private static boolean isTextEqual(DetailAST instanceOfNode, DetailAST notEqualNode) {
+        boolean error = false;
+        if (getFullText(instanceOfNode).equals(getFullText(notEqualNode))) {
```

Comment:
you can simple use `return <some_conditions>` without creating local variable,

also could you please add conditions `instanceOfNode != null && isNullTest(notEqualNode)`, so you can remove these conditions from  `checkOuterNotEqual` and `checkOuterInstanceOf`, and also this addition will help to reuse this function inside [visitToken](https://github.com/checkstyle/checkstyle/pull/8218/files#diff-f15ca0d8bd7e602e3a75f9f63d1748feR105) method

Reply:
done, finish first, and the conditions in  checkOuterNotEqual(ast) and checkOuterInstanceOf(ast) have a bit difference, so I didn't change this.